src/main/java/com/yelp/nrtsearch/server/luceneserver/search/SearchContext.java

```diff
@@ -48,6 +48,7 @@ public class SearchContext implements FieldFetchContext {
   private final List<RescoreTask> rescorers;
   private final SharedDocContext sharedDocContext;
   private final HighlightFetchTask highlightFetchTask;
+  private final Map<String, Object> highlighterContext;
```

Comment:
I suggest renaming this to something like `extraContext` or `cache` or `customContext` or something more general since even though this is called `highlighterContext` it is totally possible for any user of SearchContext to add any custom context in here. Also would be good to add a comment here saying that this is a common context that can be used by plugins otherwise someone reading this code would think that the field is unused and can be removed.

Reply:
OK. In this case, it shall be non-nullable even if the highlighter task doesn't exists.